Yoast/yoast-components
# Snippet editor improve fields focus and blur

## Summary

This PR can be summarized in the following changelog entry:

* Avoid the snippet editor fields to stay in a `isActive` state when blurred

## Relevant technical choices:

- adds an `onBlur` handler to set the `activeField` to `null` when the fields are blurred
- this required some adjustments to the way clicking on the preview elements works, it now uses `onMouseUp` instead of `onClick`, please refer to inline comment (we might want to re-think how functions are passed and bound to the snippet preview elements)
- changes `onMouseOver` to `onMouseEnter` as the former happens too often
- always focuses the fields when clicking on their labels: this was prevented by the condition `activeField !== prevActiveField`
- tries to reduce the number of times the fields get focused: this was happening under the hood also when hovering the snippet preview (and potentially it happens for any props update)
- removes the usage of lodash `partial()` for consistency and readability

## Test instructions

- run the standalone yoast-components (`yarn start` and then go to localhost:3333)
- click on an element in the preview
- verify the corresponding field in the editor gets focused
- verify that when clicking outside the preview or outside the fields, the fields don't stay in an "active" state any longer
- same when tabbing away from the fields
- verify that clicking on the preview elements always focuses the associated field correctly

Note 1: I see an issue on Safari, but it happens also on `develop`, will open a separate issue.

Note 2: there's a problem with the replaced variables within the fields in IE 11, will open a separate issue

Fixes #503
